src/data/sampler/largestTriangle3.js

```diff
@@ -0,0 +1,101 @@
+import d3 from 'd3';
+import {identity, noop} from '../../util/fn';
+
+export default function() {
+
+    var bucketSize = 10,
+        xValue = identity,
+        yValue = identity;
+
+    var largestTriangle3 = function(data) {
+
+        if (bucketSize >= data.length) {
+            return data;
+        }
+
+        var buckets = getBuckets(data);
+        var firstBucket = data[0];
+        var lastBucket = data[data.length - 1];
+
+        // Keep track of the last selected bucket info and all buckets
+        // (for the next bucket average)
+        var allBuckets = [].concat(firstBucket, buckets, lastBucket);
+
+        var lastSelectedX = xValue(firstBucket),
+            lastSelectedY = yValue(firstBucket);
+
+        var subsampledData = buckets.map(function(bucket, i) {
+
+            var highestArea = -Infinity;
+            var highestItem;
+            var nextAvgX = d3.mean(allBuckets[i + 1], xValue);
+            var nextAvgY = d3.mean(allBuckets[i + 1], yValue);
+
+            for (var j = 0; j < bucket.length; j++) {
+                var x = xValue(bucket[j]),
+                    y = yValue(bucket[j]);
+
+                var base = (lastSelectedX - nextAvgX) * (y - lastSelectedY);
+                var height = (lastSelectedX - x) * (nextAvgY - lastSelectedY);
+
+                var area = Math.abs(0.5 * base * height);
+
+                if (area > highestArea) {
+                    highestArea = area;
+                    highestItem = bucket[j];
+                }
+            }
+
+            lastSelectedX = xValue(highestItem);
+            lastSelectedY = yValue(highestItem);
+
+            return highestItem;
+        });
+
+        // First and last data points are their own buckets.
+        return [].concat(data[0], subsampledData, data[data.length - 1]);
+    };
+
+    function getBuckets(data) {
```

Comment:
This looks very similar to the mode-median getBuckets function. Can you split that out as a separate component? e.g.

```
var bucket = tfc.data.sampler.bucket().bucketSize(20);
var bucketedData = bucket(data);
```

You can then re-use this component in both of your samplers, re-binding it to expose `bucketSize`.